qiskit/quantum_info/states/utils.py

```diff
@@ -155,3 +154,52 @@ def _funm_svd(matrix, func):
     unitary1, singular_values, unitary2 = la.svd(matrix)
     diag_func_singular = np.diag(func(singular_values))
     return unitary1.dot(diag_func_singular).dot(unitary2)
+
+
+def partial_transpose(state, qargs):
+    """Return partially transposed density matrix.
+
+    Args:
+        state (DensityMatrix): the input state.
+        qargs (list): The subsystems to be transposed.
+
+    Returns:
+        DensityMatrix: The partially transposed density matrix.
+
+    Raises:
+        QiskitError: if input state is invalid.
+        QiskitError: if indices of subsystems are invalid
+
+    """
+    state = _format_state(state, validate=False)
+    n = len(state.dims())
+    lst = np.zeros(2**n, int)
+    for i in range(2**n):
+        x = 0
+        for k in qargs:
+            x = x + (((i >> (k)) % 2) * 2**k)
+        lst[i] = x
+    if not set(qargs).issubset(set(np.arange(n))):
+        raise QiskitError("Indices of subsystems to be transposed are invalid")
+    ptden = np.empty((2**n, 2**n), complex)
+    ptden[:] = np.nan
+    if isinstance(state, Statevector):
```

Comment:
I don't quite understand what a partial transpose of a vector is, do you have a definition for this? 🙂 Maybe this operation should only be supported for actual `DensityMatrix` objects, as you also mentioned in the typehint?

Reply:
Hey @Cryoris, it seems what you're saying is true. We had considered allowing input as a Statevector and converting it to a pure state density matrix, similar to partial trace code. However, if you're suggesting that it's not necessary, we can remove that option. Please let us know.

Reply:
If a user would like to do that they could always construct the DensityMatrix explicitly and then use your function:
```python
dm = DensityMatrix(state)
partial_transpose(dm, ...)
```
so I think you don't have to cover this case here 🙂